kong/tools/wrpc/proto.lua

```diff
@@ -0,0 +1,154 @@
+local pb = require "pb"
+local grpc = require "kong.tools.grpc"
+
+local grpc_new = grpc.new
+local pb_encode = pb.encode
+
+local _M = {}
+_M.__index = _M
+setmetatable(_M, _M)
+
+local wrpc_proto_name = "wrpc.wrpc"
+
+local default_proto_path = { "kong/include/", "/usr/include/" }
+
+local function parse_annotation(annotation)
+    local ret = {}
+    for kv_pair in annotation:gmatch("[^;]+=[^;]+") do
+        local key, value = kv_pair:match("^%s*(%S-)=(%S+)%s*$")
+        ret[key] = value
+    end
+    return ret
+end
+
+---@TODO: better way to do this
+-- Parse annotations in proto files with format:
+-- +wrpc: key1=val1; key2=val2; ...
+-- use key service-id and rpc-id to get IDs for service and rpc
+function _M:parse_annotations(proto_f)
+    local svc_ids = self.svc_ids
+    local rpc_ids = self.rpc_ids
+    local annotations = self.annotations
+
+    local service = ""
+    for line in proto_f:lines() do
+        local annotation = line:match("//%s*%+wrpc:%s*(.-)%s*$")
+        if annotation then
```

Comment:
we can use `if not annotation` to do a shortcut

Reply:
> we can use `if not annotation` to do a shortcut

We don't have a `break` for Lua and therefore we have to use a goto, which is what I want to aviod.

Reply:
No, in lua `goto continue` is a popular pattern, it is human friendly.